src/rospack.cpp

```diff
@@ -2068,6 +2078,9 @@ Rosstackage::writeCache()
         }
       }
     }
+#if !defined(_MSC_VER) && !defined(__MINGW32__) && !defined(WIN32)
```

Comment:
The variable is being allocated in the `else` block above which follows the following two conditions:

* `defined(_MSC_VER)`
* `defined(__MINGW32__)`

Why is this condition additionally checking for `!defined(WIN32)`?